### PR TITLE
Add share custom function settings to share custom data without defining a middleware

### PR DESCRIPTION
**Pull request by girardinsamuel**
```python
# settings.py

INERTIA_SHARE = "users.utils.share_auth_data"

# users/utils.py

def share_auth_data(request):
    if request.user.is_authenticated:
         return {"auth": json.dumps(request.user)}
    else:
         return {"auth": {}}
```

### CODE REVIEW

**Comment by girardinsamuel** [comment]
It's not working now in the test maybe because of session mocking. Not sure how to deal with that.
Also I don't know if we should put this in middleware.py or view.py. Either way we shouldn't do too much stuff inside sharing function because it's evaluated at each requests (as mentioned in Inertia official doc).

**Comment by girardinsamuel** [comment]
I have tested on my project and it's working ! @zodman do you have an idea on how to test that data has been shared in "share" key in session ? 

I was excepting to retrieve my data in the rendered response in the test, but I guess not because the session is mocked, so I guess the idea is "to check session "share" key has been indeed called" ?

File: inertia/views.py [hunk 1]
```diff
@@ -78,7 +85,7 @@ def render_inertia(request, component_name, props=None, template_name=None):
     context = _build_context(component_name, props,
                              asset_version.get_version(),
                              url=request.get_full_path())
-
+    import pdb ; pdb.set_trace()
```
Comment (zodman) [inline inertia/views.py:88]: ups!

**Review by zodman — CHANGES_REQUESTED** [review]

File: inertia/views.py (hunk 1 above)
Comment (girardinsamuel) [reply inertia/views.py:88]: Oups indeed ! I removed that.

File: inertia/views.py [hunk 2]
```diff
@@ -48,6 +49,12 @@ def render_inertia(request, component_name, props=None, template_name=None):
             "in settings.py or pass template parameter."
         )
 
+    # share custom data if any
+    share_method_path = getattr(settings, "INERTIA_SHARE", None)
```
Comment (zodman) [inline inertia/views.py:53]: please enable it for default.

**Review by zodman — CHANGES_REQUESTED** [review]
@girardinsamuel It looks a good idea!  Create a default function and add it by default. And you can overwrite by a settings ;)

**Comment by zodman** [comment]
Fix the conflicts ...

File: inertia/views.py (hunk 2 above)
Comment (girardinsamuel) [reply inertia/views.py:53]: What would be the use of the default function if it does not share any data and if we need to define a custom one in the settings to share some data ? 
Not sure to understand 🤔

Comment (zodman) [reply inertia/views.py:53]: I agreed, but its good idea had a auth object on the context ...

**Comment by girardinsamuel** [comment]
@zodman  I made the changes !

**Pull request merged**